Discuss: `onlyDisableWhiteboardAppearsInAppBarIfAllAppBarButtonsAreDisabledWithWhiteboard` test failure

This one needs thought

* If 'Toggle Auto Advance' is running, but 'Never' in the menu, should it appear?
* Other cases where menu is 'Never': should not be visible.

```
ReviewerTest > onlyDisableWhiteboardAppearsInAppBarIfAllAppBarButtonsAreDisabledWithWhiteboard FAILED
    java.lang.AssertionError: No menu items should be visible if all are disabled in Settings - Reviewer - App Bar Buttons
    Expected: an empty collection
         but: <[Toggle auto advance]>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.ichi2.anki.ReviewerTest.onlyDisableWhiteboardAppearsInAppBarIfAllAppBarButtonsAreDisabledWithWhiteboard(ReviewerTest.kt:192)
```

* https://github.com/ankidroid/Anki-Android/issues/17701
* https://github.com/ankidroid/Anki-Android/pull/18645/commits/1788e223a1165edecbaa05d35f32d81c9a5931f2

--- a/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt
@@ -882,6 +882,7 @@ open class Reviewer :
             }
         }
         menu.findItem(R.id.action_toggle_auto_advance).apply {
+            if (actionButtons.status.autoAdvanceMenuIsNeverShown()) return@apply
             // always show if enabled (to allow disabling)
             // otherwise show if it will have an effect
             isVisible = automaticAnswer.isEnabled() || automaticAnswer.isUsable()

--- a/AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.kt
@@ -115,6 +115,8 @@ class ActionButtonStatus {
 
     fun flagsIsOverflown(): Boolean = customButtons[R.id.action_flag] == SHOW_AS_ACTION_NEVER
 
+    fun autoAdvanceMenuIsNeverShown(): Boolean = customButtons[R.id.action_toggle_auto_advance] == MENU_DISABLED
+
     companion object {
         const val SHOW_AS_ACTION_NEVER = MenuItem.SHOW_AS_ACTION_NEVER
         const val SHOW_AS_ACTION_IF_ROOM = MenuItem.SHOW_AS_ACTION_IF_ROOM

